opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -111,6 +166,12 @@ def create(attributes: typing.Optional[Attributes] = None) -> "Resource":
             resource = _DEFAULT_RESOURCE
         else:
             resource = _DEFAULT_RESOURCE.merge(Resource(attributes))
+
+        default_service_name = "unknown_service"
+        process_executable_name = resource.attributes.get(PROCESS_EXECUTABLE_NAME, None)
+        if process_executable_name:
+            default_service_name += ":" + process_executable_name
+        resource = resource.merge(Resource({SERVICE_NAME: default_service_name}))
         return resource.merge(OTELResourceDetector().detect())
```

Comment:
If users pass in `service_name` through env vars, it should be used instead of the default.